bookmarks/models.py

```diff
@@ -7,11 +7,31 @@
 
 
 class Bookmark (models.Model):
-    user = models.ForeignKey(User, on_delete=models.CASCADE)
-    course = models.ForeignKey(Course, on_delete=models.CASCADE, default='SOME STRING')
+    user = models.ForeignKey(User, on_delete=models.PROTECT)
+    course = models.ForeignKey(Course, on_delete=models.PROTECT)
     url = models.URLField(max_length=200)
     urlname = models.CharField(max_length=30)
-    last_view_date = models.DateTimeField(auto_now=True, null=True)
+    last_view_date = models.DateTimeField(auto_now=True, blank=True)
 
     def __str__(self):
         return self.urlname
+
+    @classmethod
+    def add_bookmark(cls, user, course, url, urlname):
+        """Creates new bookmark"""
+        b = Bookmark.objects.create(user=user, course=course, url=url, urlname=urlname)
```

Comment:
Why use Bookmark here and cls in the function below?
Best practice is to use cls in both, but whichever way you do it, it should be the same